pkg/tasks/clustermonitoringoperator.go

```diff
@@ -72,19 +72,25 @@ func (t *ClusterMonitoringOperatorTask) Run() error {
 		return errors.Wrap(err, "reconciling Cluster Monitoring Operator ServiceMonitor failed")
 	}
 
-	s, err := t.client.GetSecret("openshift-monitoring", "grpc-tls")
-	if apierrors.IsNotFound(err) {
-		err = nil
-		s = nil // this will be a zero value if it was not found
+	s, err := t.factory.GRPCSecret()
+	if err != nil {
+		return errors.Wrap(err, "error initializing Cluster Monitoring Operator GRPC TLS secret")
 	}
 
-	if err != nil {
+	loaded, err := t.client.GetSecret(s.Namespace, s.Name)
+	switch {
+	case apierrors.IsNotFound(err):
+		// No secret was found, proceed with the default empty secret from manifests.
```

Comment:
Should we add debug logs here besides comments?

Reply:
that's a great idea :+1: